### PR TITLE
Save the language in the user's session on first visit

### PR DESCRIPTION
**Pull request by karyon**
This resolves the following inconsistency: If a user first visits the page, the brower's language gets written to user.language, but is not saved in the session. If the user now changes the browser's language, EvaP's language will also change. However, if the user logs out and logs in again, changing the browser's language will not change EvaP's language, because on the second login, the user.language will be written to the session.

@He3lixxx tripped over this while writing tests for the results cache, see #1229.

### CODE REVIEW

File: evap/staff/tests/test_views.py [hunk 1]
```diff
@@ -352,15 +353,19 @@ def setUpTestData(cls):
...
         self.assertGreater(position_evaluation1, position_evaluation2)
+        self.app.reset() # language is only loaded on login, so we're forcing a re-login here
 
-        page = self.app.get(self.url, user='manager', extra_environ={'HTTP_ACCEPT_LANGUAGE': 'de'}).body.decode("utf-8")
+        UserProfile.objects.filter(username='manager').update(language='de')
+        page = self.app.get(self.url, user='manager').body.decode("utf-8")
         position_evaluation1 = page.find("Evaluation 1")
         position_evaluation2 = page.find("Evaluation 2")
         self.assertLess(position_evaluation1, position_evaluation2)
+        translation.deactivate()
```
Comment (He3lixxx) [inline evap/staff/tests/test_views.py]: Why is this necessary?

Comment (karyon) [reply evap/staff/tests/test_views.py]: good point. I added a comment and added the same line to the other tests where it's needed.

Comment (He3lixxx) [reply evap/staff/tests/test_views.py]: The comments are good, but why do we need to reset the translation? I mean, shouldn't the app we are querying only keep the language based on the user (so sending another request with another user resets it)? Or is the issue that if the next request is not made with a specific user, it will keep this language activated? That would be quite strange, wouldn't it? I wouldn't expect the actual server to behave that way.

In general, this feels wrong to me, I would expect it work as if the translation is automatically reset after each test...

Comment (karyon) [reply evap/staff/tests/test_views.py]: You are right, this was weird and there was a fix for this: the LocaleMiddleware was after the AuthenticationMiddleware, therefore the set_or_get_language picked up whatever language was active in the thread before the LocaleMiddleware had any chance to set a new language according to the request (and, if the request does not give any hints, it would choose the default setting). Fixing the middleware order made the manual deactivations unnecessary.
